bench-vortex/src/benchmark_driver.rs

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+//! Benchmark driver that handles CLI logic and orchestrates benchmark execution
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use indicatif::ProgressBar;
+use itertools::Itertools;
+use log::warn;
+use vortex_datafusion::metrics::VortexMetricsFinder;
+
+use crate::benchmark_trait::Benchmark;
+use crate::display::DisplayFormat;
+use crate::engines::{EngineCtx, benchmark_datafusion_query, benchmark_duckdb_query};
+use crate::measurements::QueryMeasurement;
+use crate::metrics::{MetricsSetExt, export_plan_spans};
+use crate::query_bench::{filter_queries, print_results, setup_logging_and_tracing};
+use crate::utils::constants::STORAGE_NVME;
+use crate::utils::new_tokio_runtime;
+use crate::{Engine, Format, Target, df, vortex_panic};
+
+/// Configuration for the benchmark driver
+pub struct DriverConfig {
+    pub targets: Vec<Target>,
+    pub iterations: usize,
+    pub threads: Option<usize>,
+    pub verbose: bool,
+    pub display_format: DisplayFormat,
+    pub disable_datafusion_cache: bool,
+    pub queries: Option<Vec<usize>>,
+    pub exclude_queries: Option<Vec<usize>>,
+    pub output_path: Option<PathBuf>,
+    pub emit_plan: bool,
+    pub export_spans: bool,
+    pub show_metrics: bool,
+    pub hide_progress_bar: bool,
+}
+
+/// Run a benchmark using the provided implementation and configuration
+pub fn run_benchmark<B: Benchmark>(benchmark: B, config: DriverConfig) -> Result<()> {
+    let _trace_guard = setup_logging_and_tracing(
+        config.verbose,
+        &format!("{}.trace.json", benchmark.dataset_name()),
+    )?;
+
+    // Validate arguments
+    validate_args(&config)?;
+
+    // Generate data for each target (idempotent)
+    for target in &config.targets {
+        benchmark.generate_data(target)?;
+    }
+
+    let filtered_queries = filter_queries(
+        benchmark.queries()?,
+        config.queries.as_ref(),
+        config.exclude_queries.as_ref(),
+    );
+
+    let progress_bar = if config.hide_progress_bar {
+        ProgressBar::hidden()
+    } else {
+        ProgressBar::new((filtered_queries.len() * config.targets.len()) as u64)
+    };
+
+    let mut query_measurements = Vec::new();
+
+    for target in config.targets.iter() {
+        let tokio_runtime = new_tokio_runtime(config.threads);
+
+        let mut engine_ctx = benchmark.setup_engine_context(
+            target,
+            config.disable_datafusion_cache,
+            config.emit_plan,
+        )?;
+
+        // Register tables
```

Comment:
Not sure about the comments which really just echo the next function being called.